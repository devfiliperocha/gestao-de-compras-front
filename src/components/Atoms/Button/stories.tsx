import Button, { ButtonPropsBase } from '.'
import { Meta, Story } from '@storybook/react/types-6-0'
import Icon from 'components/Atoms/Icon'

export default {
  title: 'Design System/Atoms/Button',
  component: Button,
  argTypes: {
    iconLeft: { table: { disable: true } },
    iconRight: { table: { disable: true } },
    onClick: { table: { disable: true } }
  }
} as Meta

export const Basic: Story<ButtonPropsBase> = (args) => (
  <Button {...args}>BUTTON</Button>
)
export const IconLeft: Story<ButtonPropsBase> = (args) => (
  <Button iconLeft={<Icon name="AppFolder" variant="white" />} {...args}>
    ICON LEFT
  </Button>
)
export const IconRight: Story<ButtonPropsBase> = (args) => (
  <Button iconRight={<Icon name="AppFolder" variant="white" />} {...args}>
    ICON RIGHT
  </Button>
)

const defaultArgs: Pick<ButtonPropsBase, 'color' | 'variant'> = {
  color: 'primary',
  variant: 'contained'
}
Basic.args = defaultArgs
IconLeft.args = defaultArgs
IconRight.args = defaultArgs
