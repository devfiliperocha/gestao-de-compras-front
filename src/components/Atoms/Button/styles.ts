import styled from 'styled-components'
import ButtonBase from '@material-ui/core/Button'

export const Wrapper = styled(ButtonBase)``
