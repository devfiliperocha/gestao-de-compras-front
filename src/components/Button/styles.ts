import styled, { css, DefaultTheme } from 'styled-components'
import ButtonBase from '@material-ui/core/Button'

const wrapperModifiers = {
  contained: (theme: DefaultTheme) => css`
    color: ${theme.palette.common.white};
  `
}

export const Wrapper = styled(ButtonBase)`
  ${({ theme, variant }) => css`
    ${variant === 'contained' && wrapperModifiers.contained(theme)}
  `}
`
