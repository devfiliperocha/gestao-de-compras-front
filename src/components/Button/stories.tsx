import Button, { ButtonProps } from '.'
import { Meta, Story } from '@storybook/react/types-6-0'
import { PlaylistAddCheckOutlined } from '@material-ui/icons'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    leftIcon: { type: '' },
    rightIcon: { type: '' }
  }
} as Meta

export const Basic: Story<ButtonProps> = (args) => <Button {...args} />

export const IconLeft: Story<ButtonProps> = (args) => (
  <Button {...args} leftIcon={<PlaylistAddCheckOutlined />} />
)
export const IconRight: Story<ButtonProps> = (args) => (
  <Button {...args} rightIcon={<PlaylistAddCheckOutlined />} />
)

Basic.args = {
  children: 'BASIC',
  variant: 'primary'
}

IconLeft.args = {
  children: 'ICON LEFT',
  variant: 'primary'
}
IconRight.args = {
  children: 'ICON RIGHT',
  variant: 'primary'
}
