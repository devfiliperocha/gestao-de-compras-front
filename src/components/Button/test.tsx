import { PlayArrow } from '@material-ui/icons'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Button from '.'

describe('<Button/>', () => {
  it('should render primary variant by default', () => {
    renderWithTheme(<Button>Button</Button>)
    const button = screen.getByRole('button', { name: /Button/i })
    expect(button).toHaveClass('MuiButton-contained')
    expect(button).toHaveClass('MuiButton-containedPrimary')
  })

  it('should render left icon if passed', () => {
    renderWithTheme(
      <Button leftIcon={<PlayArrow data-testid="icon" />}>Button</Button>
    )
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })
  it('should render right icon if passed', () => {
    renderWithTheme(
      <Button rightIcon={<PlayArrow data-testid="icon" />}>Button</Button>
    )
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render secondary variant if passed', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)
    const button = screen.getByRole('button', { name: /Button/i })
    expect(button).toHaveClass('MuiButton-outlined')
    expect(button).toHaveClass('MuiButton-outlinedSecondary')
  })

  it('should render list variant if passed', () => {
    renderWithTheme(<Button variant="list">Button</Button>)
    const button = screen.getByRole('button', { name: /Button/i })
    expect(button).toHaveClass('MuiButton-text')
    expect(button).toHaveClass('MuiButton-textPrimary')
  })
  it('should render link variant if passed', () => {
    renderWithTheme(<Button variant="link">Button</Button>)
    const button = screen.getByRole('button', { name: /Button/i })
    expect(button).toHaveClass('MuiButton-text')
    expect(button).toHaveClass('MuiButton-textWarning')
  })
})
