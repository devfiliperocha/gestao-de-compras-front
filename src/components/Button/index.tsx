import { Button as ButtonBase, IconProps } from '@material-ui/core'
import * as S from './styles' /** S = Styles */

type ButtonBaseProps = Partial<typeof ButtonBase>

type Variants = 'primary' | 'secondary' | 'link' | 'list'

const variantsConfig: Record<Variants, ButtonBaseProps> = {
  primary: {
    color: 'primary',
    variant: 'contained',
    disableElevation: true,
    size: 'medium'
  },
  secondary: {
    color: 'secondary',
    variant: 'outlined',
    size: 'medium'
  },
  link: {
    color: 'accent',
    variant: 'text',
    size: 'medium'
  },
  list: {
    color: 'primary',
    variant: 'text',
    size: 'medium'
  }
}

export type ButtonProps = {
  children?: React.ReactNode
  variant?: Variants
  leftIcon?: IconProps
  rightIcon?: IconProps
}

const Button = ({
  children,
  variant = 'primary',
  leftIcon,
  rightIcon
}: ButtonProps) => (
  <S.Wrapper
    {...variantsConfig[variant]}
    endIcon={rightIcon}
    startIcon={leftIcon}
  >
    {!!children && <span>{children} </span>}
  </S.Wrapper>
)

export default Button
