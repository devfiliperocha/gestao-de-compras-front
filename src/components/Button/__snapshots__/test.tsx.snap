// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render the button 1`] = `
<button
  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-zddlty-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span>
    Button
     
  </span>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`
